app/javascript/components/supplemental_application/SupplementalApplicationPage.js

```diff
@@ -1,48 +1,96 @@
-      { title: application.name, link: '#' }
-    ]
+  constructor(props) {
+    super(props)
+    this.state = {
+      // A frozen copy of the application state that is currently persisted to salesforce. This is the latest saved copy.
+      persistedApplication: cloneDeep(props.application)
+    }
   }
 
-  const tabSection = {
-    items: [
-      { title: 'Short Form Application',    url: appPaths.toApplication(application.id) },
-      { title: 'Supplemental Information',  url: appPaths.toApplicationSupplementals(application.id) }
-    ]
+
+  handleSaveApplication = async (application) => {
+    const { persistedApplication } = this.state
+
+    // We clone the modified application in the UI since those are the fields we want to update
+    const synchedApplication = cloneDeep(application)
+
+    // Monthly rent and preferences are only updated in handleSavePreference below.
+    // We set this values so we keep whaterver we save in the panels
+    synchedApplication.total_monthly_rent = persistedApplication.total_monthly_rent
+    synchedApplication.preferences = cloneDeep(persistedApplication.preferences)
+
+    await updateApplicationAction(synchedApplication)
+    this.setState({ persistedApplication: synchedApplication })
   }
 
-  return (
-    <CardLayout pageHeader={pageHeader} tabSection={tabSection}>
-      <SupplementalApplicationContainer
-        application={application}
-        statusHistory={statusHistory}
-        formFields={formFields}
-        onSubmit={onSubmit}
-        fileBaseUrl={fileBaseUrl}
-      />
-    </CardLayout>
-  )
+  handleSavePreference =  async (preferenceIndex, application) => {
+    const { persistedApplication } = this.state
+
+    // We clone the lates saved copy, so we can use the latest saved fields.
+    const synchedApplication = cloneDeep(persistedApplication)
+
+    // We use the persisted copy and set only the fields updated in the panel
+    synchedApplication.total_monthly_rent = application.total_monthly_rent
+    synchedApplication.preferences[preferenceIndex] = application.preferences[preferenceIndex]
+
+    await updateApplicationAction(synchedApplication)
+    this.setState({ persistedApplication: synchedApplication })
+  }
+
+  render() {
+    const { statusHistory, fileBaseUrl, application } = this.props
+
+    const pageHeader = {
+      title: `${application.name}: ${application.applicant.name}`,
+      breadcrumbs: [
+        { title: 'Lease Ups', link: '/lease-ups' },
+        { title: application.listing.name, link: appPaths.toListingLeaseUps(application.listing.id) },
+        { title: application.name, link: '#' }
+      ]
+    }
+
+    const tabSection = {
+      items: [
+        { title: 'Short Form Application',    url: appPaths.toApplication(application.id) },
+        { title: 'Supplemental Information',  url: appPaths.toApplicationSupplementals(application.id) }
+      ]
+    }
+
+    const context = {
```

Comment:
@akegan @callaghanc basically here we pass to the context some var from the state, the props and some action that can be use down the tree that will update this state. (one way data flow)